src/main/kotlin/ch/srgssr/pillarbox/monitoring/event/model/EventRequest.kt

```diff
@@ -64,6 +64,15 @@ private class DataDeserializer : JsonDeserializer<Any?>() {
         .build()
   }
 
+  fun isHackerOrRobot(userAgent: UserAgent): Boolean =
+    listOf("DeviceClass", "LayoutEngineClass", "AgentClass", "AgentSecurity")
+      .any { field ->
+        userAgent.getValue(field)?.let { value ->
+          value.startsWith("Hacker", ignoreCase = true) ||
+            value.startsWith("Robot", ignoreCase = true)
+        } ?: false
+      }
```

Comment:
Would the following work?
```suggestion
  fun isHackerOrRobot(userAgent: UserAgent): Boolean =
    !UserAgentClassifier.isHuman(userAgent)
```

Reply:
This function only checks `DevicClass`, but according to the [documentation](https://yauaa.basjes.nl/expect/fieldvalues/index.html) it is not guaranteed that this field will be always available:

> Note that **not all fields are always available**. So if you look at a specific field you will in general find null values and “Unknown” in there as well.

Reply:
Let's keep your solution then 👍🏻